design/codebases.go

```diff
@@ -179,6 +190,37 @@ var _ = a.Resource("codebase", func() {
 		a.Response(d.NotFound, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+	a.Action("cheState", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.GET("/che/state"),
+		)
+		a.Description("Get che server state.")
+		a.Response(d.OK, func() {
+			a.Media(cheServerState)
+		})
+		a.Response(d.BadRequest, JSONAPIErrors)
+		a.Response(d.InternalServerError, JSONAPIErrors)
+		a.Response(d.NotFound, JSONAPIErrors)
```

Comment:
All errors are treated as Internal, why do you specify NotFound ? Could we get or identify `NotFound` errors from Che ? In that way we could propagate NotFound errors if the namespace doesn't exist.